regolith/errors.go

```diff
@@ -0,0 +1,161 @@
+	fileWriteError = "Failed to write to file.\nPath: %s"
+
+	// Error used when program fails to parse JSON file
+	jsonUnmarshalError = "Failed to parse JSON.\nPath: %s"
+
+	// Error used when Regolith fails to parse a property os JSON
+	jsonPropertyParseError = "Failed to parse JSON property.\nProperty: %s"
+
+	// Error used when Regolith expects a property but it's missing
+	jsonPropertyMissingError = "Required JSON property is missing.\nProperty: %s"
+
+	// Errror used when JSON property is not an expected type
+	jsonPropertyTypeError = "JSON property has unexpected type." +
+		"\nProperty: %s\nExpected: %s"
+
+	// Error used when Regolith fails to parse a property os JSON
+	jsonPathParseError = "Failed to parse JSON.\nJSON Path: %s"
+
+	// Error used when JSON path is missing
+	jsonPathMissingError = "Required JSON path is missing.\nJSON Path: %s"
+
+	// Error used when JSON path exists but the type is wrong
+	jsonPathTypeError = "Invalid data type.\nJSON Path: %s\n" +
+		"Expected type: %s"
+
+	// Error used when RunSubProcess funciton fails
+	runSubProcessError = "Failed to run sub process."
+
+	// Error used when safe mode is when user tries to perform an unsafe
+	// operation with safe mode enabled
+	safeModeEnabledError = "Safe mode is enabled. " +
+		"It's a feature that protects you from running unsafe code.\n" +
+		"You can turn it off using command:\nregolith unlock"
+
+	// Error used when remote filter fails to access its subfilter collection.
+	// The error doesn't print the name of the filter because the
+	// subfilterCollection method is private and it's always a part of some
+	// other, higher level action which provides that information when it
+	// fails.
+	remoteFilterSubfilterCollectionError = "Failed to list subfilters."
+
+	// Error used when GetRemoteFilterDownloadRef function fails
+	getRemoteFilterDownloadRefError = "Failed to get download link for the filter.\n" +
+		"Filter repository Url: %s\n" +
+		"Filter name: %s\n" +
+		"Filter version: %s"
+
+	// Error used when CreateFilterRunner method of FilterInstaller fails
+	createFilterRunnerError = "Failed to create filter runner.\nFilter: %s"
+
+	// Warning used when Git is not installed
+	gitNotInstalledWarning = "Git is not installed. Git is required to download " +
+		"filters.\n You can download Git from https://git-scm.com/downloads"
+
+	// Error used when filterFromObject function fails
+	filterFromObjectError = "Failed to parse filter from JSON object."
+
+	// Error used when remote filter fails to download
+	remoteFilterDownloadError = "Failed to download filter.\nFilter: %s"
+
+	// Error used when exec.Command fails.
+	execCommandError = "Failed to execute command.\nCommand: %s"
+
+	// Error used when FilterRunner.Check method fails
+	filterRunnerCheckError = "Filter check failed.\nFilter: %s"
+
+	// Error used when certain function is not implemented on this system
+	notImplementedOnThisSystemError = "Not implemented for this system."
```

Comment:
Could we get some more info here? I liked the old message. which included the runtime name. 

`"Unsupported operating system: '%s'", runtime.GOOS)`

Reply:
Why do you think it's important to tell people what OS they're using?